src/lib/repositories_v2/impl/chef_repositoty_impl.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime
+
+from src.constants.audit import Status
+from src.core.ioc import get_ioc_instance
+from src.lib.entities.sqlalchemy_orm_mapping import Chef, Order
+from src.lib.repositories_v2.chef_repository import ChefRepository
+
+
+class ChefRepositoryImpl(ChefRepository):
+    def __init__(self):
+
+        ioc = get_ioc_instance()
+        self.session = ioc.get_instance("orm_session")
+
+    def add(self, chef):
+        chef.created_date = datetime.now()
+        chef.updated_by = chef.created_by
+        chef.updated_date = chef.created_date
+        self.session.add(chef)
+        self.session.commit()
+
+    def get_by_id(self, chef_id):
+        return (
+            self.session.query(Chef)
+            .filter(Chef.id == chef_id)
+            .filter(Chef.entity_status == "ACTIVE")
+            .first()
+        )
+
+    def get_all(self):
+        chefs = self.session.query(Chef).filter(Chef.entity_status == "ACTIVE")
+        return chefs
+
+    def delete_by_id(self, chef_id, chef):
+        chef_to_be_delete = self.session.query(Chef).filter(Chef.id == chef_id).first()
+        chef_to_be_delete.entity_status = Status.DELETED.value
+        chef_to_be_delete.updated_date = datetime.now()
+        chef_to_be_delete.updated_by = chef.updated_by
+        self.session.add(chef_to_be_delete)
+        self.session.commit()
+
+    def update_by_id(self, chef_id, chef):
+        chef_to_be_updated = self.session.query(Chef).filter(Chef.id == chef_id).first()
+        chef_to_be_updated.name = chef.name or chef_to_be_updated.name
+        chef_to_be_updated.description = (
+            chef.description or chef_to_be_updated.description
+        )
+        chef_to_be_updated.updated_date = datetime.now()
+        chef_to_be_updated.updated_by = chef.updated_by
+        self.session.add(chef_to_be_updated)
+        self.session.commit()
+
+    def get_available_chefs(self):
+        assigned_chef_ids = self.session.query(Order.assigned_chef_id).filter(
```

Comment:
Could you do this in a single query? It is not a good idea to make 2 unnecessary calls to the database. 1 should be enough for this operation

Reply:
done.